staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation_test.go

```diff
@@ -698,17 +773,16 @@ func TestValidationExpressions(t *testing.T) {
 				"something": intOrStringType(),
 			}),
 			valid: []string{
-				// typical int-or-string usage would be to check both types
-				"type(self.something) == int ? self.something == 1 : self.something == '25%'",
-				// to require the value be a particular type, guard it with a runtime type check
+				// In Kubernetes 1.24 and later, the CEL type returns false for an int-or-string comparison against the
+				// other type, making it safe to write validation rules like:
+				"self.something == '25%'",
+				"self.something != 1",
+				"self.something == 1 || self.something == '25%'",
+
+				// In Kubernetes 1.23 and earlier, all int-or-string access must be guarded by a type check to prevent
+				// a runtime error attempting an equality check between string and int types.
 				"type(self.something) == string && self.something == '25%'",
-			},
-			errors: map[string]string{
-				// because the type is dynamic type checking fails a runtime even for unrelated types
-				"self.something == ['anything']": "no such overload",
```

Comment:
is this not even possible to get past type checking any more?

Reply:
I really shouldn't have dropped this test case entirely. I've added this test to "valid" instead:

```
// Because the type is dynamic it receives no type checking, and evaluates to false when compared to
// other types at runtime.
"self.something != ['anything']",
```